crates/core/src/repofile.rs

```diff
@@ -10,9 +12,41 @@ pub(crate) mod snapshotfile;
 pub trait RepoFile: Serialize + DeserializeOwned + Sized + Send + Sync + 'static {
     /// The [`FileType`] associated with the repository file
     const TYPE: FileType;
+    /// The Id type associated with the repository file
+    type Id: From<Id> + Send;
+}
+
+/// Marker trait for Ids which identify repository files
+pub trait RepoId: Deref<Target = Id> + From<Id> + Sized + Send + Sync + 'static {
+    /// The [`FileType`] associated with Id type
+    const TYPE: FileType;
+}
+
+#[macro_export]
+/// Generate newtypes for `Id`s identifying Repository files
+macro_rules! new_repoid {
+    ($a:ident, $b: expr) => {
+        $crate::new_id!($a, concat!("repository file of type", stringify!($b)));
+        impl $crate::repofile::RepoId for $a {
+            const TYPE: FileType = $b;
+        }
+    };
+}
+
+#[macro_export]
```

Comment:
Would name it `impl_repo_file` to make us aware, that it just adds an implementation for RepoFile

Reply:
renamed.